mlir/test/Transforms/loop-invariant-code-motion.mlir

```diff
@@ -929,3 +929,24 @@ func.func @speculate_dynamic_pack_and_unpack(%source: tensor<?x?xf32>,
   }
   return
 }
+
+// -----
+
+// CHECK-LABEL: func @hoist_from_scf_while(
+//   CHECK-DAG:   arith.constant 1 : i32
+//   CHECK-DAG:   arith.constant 10 : i32
+//       CHECK:   scf.while
+//   CHECK-NOT:     arith.constant
```

Comment:
Not sure whether its worth changing, but hoisting `arith.constant` isn't really interesting for LICM given that `canonicalize` is capable of hoisting them as well. Maybe more complicated pure ops like `arith.cmpi` or `arith.addi` should be tested instead.